### PR TITLE
Sim/Vol dtype passthrough

### LINKED ISSUE #793: Using Synth Volumes with Simulation crashes
Tried to use our new volume.  The following code fails with incorrect dtypes.

```
...
        self.sim = Simulation(
            vols=AsymmetricVolume(32,1).generate(),
            n=1024,
            offsets=0,
        )
``` 

```
        if self.vols.dtype != self.dtype:
>           raise RuntimeError(
                f"{self.__class__.__name__} dtype {self.dtype}"
                f" does not match provided vols.dtype {self.vols.dtype}."
            )
E           RuntimeError: Simulation dtype float32 does not match provided vols.dtype float64.
```

I believe we want to pass through here. Will push a patch.

### PR DESCRIPTION
**Pull request by garrettwrong**
I think this will Close #793 , but I want to add a unit test for it... and maybe reword the comments.

### CODE REVIEW

**Review by chris-langfield — APPROVED** [review]
This looks good thanks

**Comment by garrettwrong** [comment]
Cool, I'm going to merge it so I can use it in another PR. Thanks.
